dcm4che-core/src/main/java/org/dcm4che3/io/stream/DelegateImageInputStream.java

```diff
@@ -0,0 +1,264 @@
+//
+///////////////////////////////////////////////////////////////
+//                C O P Y R I G H T  (c) 2019                //
+//        Agfa HealthCare N.V. and/or its affiliates         //
+//                    All Rights Reserved                    //
+///////////////////////////////////////////////////////////////
+//                                                           //
+//       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF      //
+//        Agfa HealthCare N.V. and/or its affiliates.        //
+//      The copyright notice above does not evidence any     //
+//     actual or intended publication of such source code.   //
+//                                                           //
+///////////////////////////////////////////////////////////////
+//
+package org.dcm4che3.io.stream;
+
+import javax.imageio.stream.IIOByteBuffer;
+import javax.imageio.stream.ImageInputStream;
+import java.io.IOException;
+import java.nio.ByteOrder;
+
+/**
+ * Delegate to an underlying ImageInputStream.
+ * @author Andrew Cowan (andrew.cowan@agfa.com)
+ */
+public class DelegateImageInputStream implements ImageInputStream {
```

Comment:
To support the passing of an external IIS into a DicomImageReader, we provide the capability of wrapping that IIS in a delegate that keeps it open.

Reply:
The name is confusing. For me a "delegate" is the inner field ("iis") that you are delegating to. So the wrapping class should not also be called "delegate". Maybe better: "DelegatingImageInputStream"? Or "NotCloseableImageInputStreamWrapper"?